src/spark_manager.py

```diff
@@ -143,16 +151,229 @@ def create_cluster(self,
+                    "app": "spark",
+                    "component": "master",
+                    "user": self.username,
+                    "cluster-id": self.cluster_id,
+                },
+            },
+            "spec": {
+                "replicas": 1,
+                "selector": {
+                    "matchLabels": {
+                        "app": "spark",
+                        "component": "master",
+                        "user": self.username,
+                        "cluster-id": self.cluster_id,
+                    }
+                },
+                "template": {
+                    "metadata": {
+                        "labels": {
+                            "app": "spark",
+                            "component": "master",
+                            "user": self.username,
+                            "cluster-id": self.cluster_id,
+                        }
+                    },
+                    "spec": {
+                        "hostname": f"spark-master-{self.username.lower()}",
+                        "containers": [
+                            {
+                                "name": "spark-master",
+                                "image": self.image,
+                                "imagePullPolicy": self.image_pull_policy,
+                                "ports": [
+                                    {
+                                        "containerPort": self.DEFAULT_MASTER_PORT,
+                                        "name": "master-comm",
+                                        "protocol": "TCP",
+                                    },
+                                    {
+                                        "containerPort": self.DEFAULT_MASTER_WEBUI_PORT,
+                                        "name": "master-ui",
+                                        "protocol": "TCP",
+                                    },
+                                ],
+                                "env": env,
+                                "resources": {
+                                    "requests": {"memory": memory, "cpu": str(cores)}
+                                    # TODO: Add resource limits if needed
+                                },
+                            }
+                        ],
+                    },
+                },
+            },
+        }
+
+        try:
+            # Try to create the deployment first
+            self.apps_api.create_namespaced_deployment(
+                namespace=self.namespace, body=deployment
+            )
+            logger.info(f"Created Spark master deployment: {self.master_name}")
+        except ApiException as e:
+            if e.status == 409:  # Conflict - already exists
+                try:
+                    # Delete the existing deployment
+                    self.apps_api.delete_namespaced_deployment(
+                        name=self.master_name, namespace=self.namespace
+                    )
```

Comment:
If it already exists is it ok to just use it?

Reply:
That makes sense. Is there a way this code could be run while an existing cluster is running a job, killing the job?

Reply:
yea. It could be. I don't think it's our top concern now - as users will trigger this creation of spark clustering. Maybe we could add a warning - but that's not a trivial task as we need to verify if there are jobs currently running within user's clustering (an idle spark session is also considered as a running session/job but spark.stop() is called explicitly).

Reply:
I'd suggest creating an issue to remind us to handle this case later

Reply:
Checking if a cluster exists and warning the user that it'll be destroyed, killing any jobs, if they continue seems enough

Reply:
hmmm I can create a ticket - it's an API call, I can warn users. But I don't know a good way to pause the execution and then wait for user's action to continue.

Reply:
https://kbase-jira.atlassian.net/browse/CDM-107

Reply:
:+1: